src/main/java/org/folio/oaipmh/ResponseConverter.java

```diff
@@ -160,9 +162,13 @@ public Object bytesToObject(byte[] byteSource) {
         jaxbUnmarshaller.setSchema(oaipmhSchema);
       }
       return jaxbUnmarshaller.unmarshal(inputStream);
-    } catch (JAXBException | IOException e) {
+    }  catch (JAXBException | IOException e) {
+      var errorMessage = "The byte array cannot be converted to JAXB object response.";
+      if (e instanceof UnmarshalException ue && ue.getLinkedException() instanceof SAXParseException se) {
```

Comment:
Can this validation be extracted method checkParsingException()?

Reply:
Good catch, fixed.